src/ctap/ctap1.rs

```diff
@@ -131,30 +137,29 @@ impl TryFrom<&[u8]> for U2fCommand {
         // +-----+-----+----+----+-----+-----+-----+
         // | CLA | INS | P1 | P2 | Lc1 | Lc2 | Lc3 |
         // +-----+-----+----+----+-----+-----+-----+
-        if apdu[0] != Ctap1Command::CTAP1_CLA {
+        if apdu.header.cla != Ctap1Command::CTAP1_CLA {
             return Err(Ctap1StatusCode::SW_CLA_NOT_SUPPORTED);
         }
 
-        let lc = (((apdu[4] as u32) << 16) | ((apdu[5] as u32) << 8) | (apdu[6] as u32)) as usize;
-
         // Since there is always request data, the expected length is either omitted or
         // encoded in 2 bytes.
-        if lc != payload.len() && lc + 2 != payload.len() {
+        // Todo: support extended APDUs now that the new parser can work with those
+        if apdu.lc as usize != apdu.data.len() && (apdu.lc as usize) + 2 != apdu.data.len() {
```

Comment:
Same comment for `TODO` style. Also, maybe set `let lc = apdu.lc as usize` here once and the some old code runs as before?

Reply:
@kaczmarczyck good idea! moved to a local `lc` variable to remove redundant casting